app/models/user.rb

```diff
@@ -23,4 +27,44 @@ def list_following
     return search.following
   end
 
+  def username
+    search = GithubInfoSearch.new(token)
+    return search.username
+  end
+
+  def potential_friends
+    following_followers = (self.list_following.map(&:name) + self.list_followers.map(&:name)).uniq
+    users_username = User.with_token.map do |user|
+      user.username
+    end
+    result = User.with_token.reduce({}) do |acc,user|
+      if following_followers.include?(user.username)
+        acc[user.username] = user.id
+      end
+      acc
+    end
+    result
+  end
+
+
+  #   current_username = self.username
+  #   @potential_friends ||= User.with_token.find_all do |user|
+  #     user.list_following.map(&:name).include?(current_username) || user.list_followers.map(&:name).include?(current_username)
+  #   end
+  #   result = @potential_friends.reduce({}) do |acc, friend|
+  #     acc[friend.username] = friend.id
+  #     acc
+  #   end
+  #   return result
+  # end
```

Comment:
I will remove this comment, sorry about that

Reply:
It's all good - I was about to comment this x)